lib/features/composer/presentation/composer_controller.dart

```diff
@@ -219,6 +228,20 @@ class ComposerController extends BaseController {
     }
     popBack();
   }
+
+  void _listenBrowserTabRefresh() {
+    html.window.onBeforeUnload.listen((event) async {
+      final userProfile = mailboxDashBoardController.userProfile.value;
+      _removeEmailCacheOnWebInteractor.execute();
+      if (userProfile != null) {
+        final draftEmail = await _generateEmail(
+          mailboxDashBoardController.mapDefaultMailboxId,
+          userProfile,
+        );
+        _saveEmailCacheOnWebInteractor.execute(draftEmail);
```

Comment:
IMO, you should wait for the `_removeEmailCacheOnWeb` process to complete before proceeding with the `_saveEmailCacheOnWeb` process

Reply:
_removeEmailCacheOnWeb just is void function no need to wait it